src/pyodide-py/pyodide/console.py

```diff
@@ -1,8 +1,40 @@
-from typing import Optional, Callable
+from typing import Optional, Callable, Any
 import code
 import io
 import sys
 import platform
+from contextlib import contextmanager
+import builtins
+
+# this import can fail when we are outside a browser (e.g. for tests)
+try:
+    import js
+
+    dummy_promise = js.Promise.resolve()
+    load_packages_from_imports = js.pyodide.loadPackagesFromImports
+
+except ImportError:
+
+    class FakePromise:
+        """A promise that mimic the JS promises.
+
+        Only `then is supported` and there is no asynchronicity.
+        execution occurs when then is call.
+
+        This is mainly to fake `load_packages_from_imports`
+        and `InteractiveConsole.run_complete` in contexts
+        where JS promises are not available (tests)."""
+
+        def __init__(self, args=None):
+            self.args = (args,) if args is not None else ()
+
+        def then(self, func, *args):
+            return FakePromise(func(*self.args))
+
+    dummy_promise = FakePromise()
+
+    def load_packages_from_imports(*args):
```

Comment:
Same I'm not sure if this needs to be public for now, otherwise,
```suggestion
    def _load_packages_from_imports(*args):
```

Reply:
OK.